src/main.ts

```diff
@@ -67,13 +69,18 @@ async function run() {
       if (!repository) {
         core.setFailed("❌ Expecting repository metadata.")
         return;
-      }
+      } else
       if (bodyContains || bodyDoesNotContain) {
         const PRBody = pull_request?.body;
         core.info("Checking body contents");
-        if (!PRBody) {
-          core.warning("⚠️ The PR body is empty, skipping checks");
-        } else {
+        if (!PRBody && allowEmpty) {
+            core.warning("⚠️ The PR body is empty, skipping checks");
+          
+        } else if(!PRBody && !allowEmpty){
```

Comment:
Additionally, both `if` have the same first term, `!PrBody`.Have you considered DRYing it to a single if with that condition, and an indented if-then-else with the other two?